docs/concepts/validation_decorator.md

```diff
@@ -39,151 +39,84 @@ except ValidationError as exc:
+        return d1 > d2
 
-# note: this_dir is a string here
-this_dir = os.path.dirname(__file__)
 
-print(find_file(this_dir, '^validation.*'))
-print(find_file(this_dir, '^foobar.*', max=3))
+d1 = '2000-01-01'  # (2)!
+d2 = date(2001, 1, 1)
+greater_than(d1, d2, include_equal=True)
 ```
 
-A few notes:
+1. Because `include_equal` has no type annotation, it will be inferred as [`Any`][typing.Any].
+2. Although `d1` is a string, it will be converted as a [`date`][datetime.date] object.
 
-- Though they're passed as strings, `path` and `regex` are converted to a `Path` object and regex respectively
-    by the decorator.
-- `max` has no type annotation, so will be considered as `Any` by the decorator.
+Type coercion like this can be extremely helpful, but also confusing or not desired. [Strict mode](strict_mode.md)
+can be enabled by using a [custom configuration](#custom-configuration).
 
-Type coercion like this can be extremely helpful, but also confusing or not desired.
-See [Coercion and strictness](#coercion-and-strictness) for a discussion of `@validate_call`'s limitations
-in this regard.
+!!! note "Validating the return value"
+    By default, the return value of the function is **not** validated. To do so, the `validate_return` argument
+    of the decorator can be set to `True`.
 
 ## Function signatures
 
-The `@validate_call` decorator is designed to work with functions using all possible parameter configurations and
-all possible combinations of these:
-
-* Positional or keyword arguments with or without defaults.
-* Variable positional arguments defined via `*` (often `*args`).
-* Variable keyword arguments defined via `**` (often `**kwargs`).
-* Keyword-only arguments: arguments after `*,`.
-* Positional-only arguments: arguments before `, /` (new in Python 3.8).
-
-To demonstrate all the above parameter types:
-
-```py
-from pydantic import validate_call
-
-
-@validate_call
-def pos_or_kw(a: int, b: int = 2) -> str:
-    return f'a={a} b={b}'
-
-
-print(pos_or_kw(1))
-#> a=1 b=2
-print(pos_or_kw(a=1))
-#> a=1 b=2
-print(pos_or_kw(1, 3))
-#> a=1 b=3
-print(pos_or_kw(a=1, b=3))
-#> a=1 b=3
-
-
-@validate_call
-def kw_only(*, a: int, b: int = 2) -> str:
-    return f'a={a} b={b}'
-
+The [`validate_call()`][pydantic.validate_call] decorator is designed to work with functions
+using all possible [parameter configurations][parameter] and all possible combinations of these:
 
-print(kw_only(a=1))
-#> a=1 b=2
-print(kw_only(a=1, b=3))
-#> a=1 b=3
+* Positional or keyword parameters with or without defaults.
+* Keyword-only parameters: parameters after `*,`.
+* Positional-only parameters: parameters before `, /`.
+* Variable positional parameters defined via `*` (often `*args`).
+* Variable keyword parameters defined via `**` (often `**kwargs`).
 
+!!! note "[`Unpack`][typing.Unpack] for keyword parameters"
+    [`Unpack`][typing.Unpack] and typed dictionaries can be used to annotate the variable
+    keyword parameters of a function:
```

Comment:
Maybe this even deserves its own section?

Could you also add some docs for `typing.Unpack` in the API section of the docs?
See https://github.com/pydantic/pydantic/pull/10416#issuecomment-2364225636

Reply:
> Maybe this even deserves its own section?

unpacked typed dicts are pretty niche, so I don't really know if it is necessary? And I feel it fits nicely with the last parameter type being described:

```
* Variable keyword parameters defined via `**` (often `**kwargs`).
```

Reply:
> unpacked typed dicts are pretty niche, so I don't really know if it is necessary? And I feel it fits nicely with the last parameter type being described:

That's fine, but I do think we should add some more docs for `typing.Unpack` pre v2.10. I can do that, given that you worked on the PR with the functionality.

Reply:
Regarding the addition in API docs, it could be added in the stdlib types but I don't think it really makes sense:

`Unpack` is not a type per se, it is a special form having a different meaning depending on the context. It was originally introduced for typevar tuples which we don't support as of today.

Unless your thinking about another section of the API docs?